src/projections.jl

```diff
@@ -439,37 +546,47 @@ References:
 [1]. [Differential properties of Euclidean projection onto power cone]
 (https://link.springer.com/article/10.1007/s00186-015-0514-0), Prop 2.2
 """
-function _solve_system_pow_cone(v::AbstractVector{T}, s::MOI.PowerCone; max_iters=10_000, tol=1e-10) where {T}
+function _solve_system_power_cone(
+    v::AbstractVector{T},
+    s::MOI.PowerCone;
+    max_iters_newton = DEFAULT_POWER_CONE_MAX_ITERS_NEWTON,
+    max_iters_bisection = DEFAULT_POWER_CONE_MAX_ITERS_BISSECTION,
+    tol=DEFAULT_POWER_CONE_TOL_CONV,
+) where {T}
     x, y, z = v
     α = s.exponent
-    Phi_prod(xi, αi, z, r) = max(xi + sqrt(xi^2 + 4*αi*r*(abs(z) - r)), 1e-12)
-    Phi(r) = 0.5*(Phi_prod(x,α,z,r)^α * Phi_prod(y,1-α,z,r)^(1-α)) - r
-    Phi(r,px,py) = 0.5*(px^α * py^(1-α)) - r
-    dPhi_prod_dr(xi,αi,z,r,px) = 2*αi*(abs(z) - 2r) / (px - xi)
-    dPhi_dr(r,phi,px,py,dpx,dpy) = (phi+r) * (α * dpx / px + (1-α) * dpy / py) - 1
-
-    # Solve with Newton method
-    # Start Newton at |z|/2. Sol in set (0, |z|)
-    px, py = zero(T), zero(T)
-    r = abs(z) / 2
-    for ii in 1:max_iters
-        px = Phi_prod(x, α, z, r)
-        py = Phi_prod(y, 1-α, z, r)
-        phi = Phi(r, px, py)
-
-        abs(phi) < tol && break
-
-        dpx = dPhi_prod_dr(x, α, z, r, px)
-        dpy = dPhi_prod_dr(y, 1-α, z, r, py)
-        dphi = dPhi_dr(r, phi, px, py, dpx, dpy)
-
-        # Newton step, bounded to interval
-        r = min(max(r - phi/dphi, 0), abs(z))
-
-        ii == max_iters && @warn("Maximum iterations hit on power cone projection")
+    # We first try to quickly get an accurate solution with Newton:
+    r_newton = _solve_system_power_cone_newton(v, s; max_iters = max_iters_newton, tol)
+    Φ_newton = !isnan(r_newton) ? _power_cone_system(r_newton, x, y, z, α) : NaN
+    # When the optimal solution `r` is close to the boundaries `0` and `|z|`,
+    # Newton has a tendency to overshoot and hit the boundary. Once the boundary
+    # is hit, it will compute `NaN` so we'll need bisection to get an answer.
+
+    Φ_bisection = Inf
+    if isnan(r_newton) || abs(Φ_newton) > tol
+        r_bisection = _solve_system_power_cone_bisection(v, s; max_iters = max_iters_bisection, tol)
+        println(r_bisection)
+        Φ_bisection = _power_cone_system(r_bisection, x, y, z, α)
+        if abs(Φ_bisection) > tol
+            # This happens for instance for
+            # `_solve_system_power_cone([-10, 10, 1e-3], MOI.PowerCone(0.15))`
```

Comment:
An example like this used to throw the warning because Newton was just returning `0` or `abs(z)`. I'm surprised the tests were actually passing. The bisection does much better on these so we should have tests that fail for these and pass thanks to this PR